.travis.yml

```diff
@@ -20,3 +20,22 @@ script: sh bin/travis-run.sh
 # the new trusty images of Travis cause build errors with psycopg2, see https://github.com/travis-ci/travis-ci/issues/8897
 dist: trusty
 group: deprecated-2017Q4
+
+stages:
+  - Flake8
+  - test
+
+jobs:
+  include:
+    - stage: Flake8
+      env: FLAKE8=True
+      install:
+        - bash bin/travis-build.bash
+        - pip install flake8==3.5.0
+        - pip install pycodestyle==2.3.0
+      script:
+        - flake8 --version
+        # stop the build if there are Python syntax errors or undefined names
+        - flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics --exclude ckan
+        # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
+        - flake8 . --count --max-line-length=127 --statistics --exclude ckan
```

Comment:
This is the key decision here - that you removed `--exit-zero` compared to CKAN core, so ckanext-harvest is being more hard-line about style than CKAN core.
 
I support this - there's a trend towards this in python these days (see popularity of the [black](https://github.com/ambv/black) and automating this aspect of code review, the idea that there is no shame when you're being kept in line by a bot, rather than a human) and we have this opportunity to start from zero errors, so why not. CKAN core should probably move towards this too.